scheduler/pkg/kafka/pipeline/kafkamanager.go

```diff
@@ -116,7 +134,10 @@ func (km *KafkaManager) createProducer() error {
 	if err != nil {
 		return err
 	}
-	km.logger.Infof("Creating producer with config %v", producerConfigMap)
+
+	configWithoutSecrets := config.WithoutSecrets(producerConfigMap)
```

Comment:
this pattern of casting `...ConfigMap` into safe format without secrets before logging repeats all over the place - potentially a convenience function that takes logger and `...ConfigMap` could be useful to skip repetitions

Reply:
Fixed in https://github.com/SeldonIO/seldon-core/pull/4966/commits/1dd883adcfbcf70874337f7729c85ea42b975a0d

Reply:
We don't know what sort of config a map is for (producer, consumer, admin?), and I think that, although it's a bit annoying, it's probably easier not to try to come up with a pattern for this just yet